trace-utils/src/send_with_retry/mod.rs

```diff
@@ -171,6 +175,24 @@ async fn send_request(
     payload: Bytes,
     http_proxy: Option<&str>,
 ) -> Result<Response<Body>, RequestError> {
+    #[cfg(feature = "compression")]
+    let req = {
+        let result = (|| -> std::io::Result<Vec<u8>> {
+            let mut encoder = Encoder::new(Vec::new(), 6)?;
```

Comment:
Just curious how you ended up deciding on level `6` here? The Go trace-agent uses zstd best-speed (which I believe is level `1`

Reply:
We can use 1 for traces, 6 is what we use for logs and I've kinda just kept using it because I find it to fit really well for our use case. Can make it optional, but then we either have to pass a struct around or an argument.

Reply:
Made it a parameter